src/DDTrace/Integrations/LaravelProvider.php

```diff
@@ -0,0 +1,137 @@
+ *        'System\ServiceProvider',
+ *   ]),
+ */
+class LaravelProvider extends ServiceProvider
+{
+    public function register()
+    {
+        if (!extension_loaded('ddtrace')) {
+            trigger_error('ddtrace extension required to load Laravel integration.', E_USER_WARNING);
+            return;
+        }
+
+        if (php_sapi_name() == 'cli') {
+            return;
+        }
+
+        // Creates a tracer with default transport and default encoders
+        $tracer = new Tracer(new Http(new Json()));
+
+        // Sets a global tracer (singleton). Also store it in the Laravel
+        // container for easy Laravel-specific use.
+        GlobalTracer::set($tracer);
+        $this->app->instance('datadog.tracer', $tracer);
+
+        // Trace middleware
+        dd_trace(Pipeline::class, 'through', function ($pipes) {
+            foreach ($pipes as $pipe) {
+                dd_trace($pipe, 'handle', function (...$args) {
+                    $scope = GlobalTracer::get()->startActiveSpan('laravel.middleware');
+                    $span = $scope->getSpan();
+                    $span->setResource(get_class($this));
+
+                    try {
+                        return $this->handle(...$args);
+                    } catch (\Exception $e) {
+                        $span->setError($e);
+                        throw $e;
+                    } finally {
+                        $scope->close();
+                    }
+                });
+            }
+            return $this->through($pipes);
+        });
+
+        // Create a trace span for every template rendered
+        // public function get($path, array $data = array())
+        dd_trace(CompilerEngine::class, 'get', function ($scope, $path, $data) {
+            $scope = GlobalTracer::get()->startActiveSpan('laravel.view');
+
+            try {
+                return $this->getModels($builder);
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // Create a span that starts from when Laravel first boots (public/index.php)
+        $scope = $tracer->startActiveSpan('laravel.request', ['start_time' => fromMicrotime(LARAVEL_START)]);
+        $scope->getSpan()->setTag(Tags\SERVICE_NAME, $this->getAppName());
+
+        // Name the scope when the route matches
+        $this->app['events']->listen(RouteMatched::class, function (RouteMatched $event) use ($scope) {
+            $span = $scope->getSpan();
+            $span->setResource($event->route->getActionName() . ' ' . Route::currentRouteName());
+            $span->setTag('laravel.route.name', Route::currentRouteName());
+            $span->setTag('laravel.route.action', $event->route->getActionName());
+            $span->setTag('http.method', $event->request->method());
+            $span->setTag('http.url', $event->request->path());
```

Comment:
Why we store only the path (`/some/relative/patch`) in a tag named `http.url` instead of the `$event->request->url()` `https://example.com/some/relative/path`?

Reply:
I think I was following an example from dogweb traces here, but happy either way. I'll change it for now.